### ISSUE TITLE
New prop pages - consider adding a checkbox to generate a binlog on build

### ISSUE THREAD

**Issue by cartermp** [dotnet/project-system#7183]
This would likely live under Build/General in the new property pages. I propose a simple checkbox:

- [ ] Generate MSBuild Binary Log on build

It could have the `?` icon pointing to [troubleshooting docs](https://docs.microsoft.com/en-us/visualstudio/ide/msbuild-logs?view=vs-2019) and have some descriptive text, like:

"Generates an MSBuild Binary Log (binlog) on build, a file that contains detailed diagnostic information about your build. It is helpful for diagnosing problems."

**Comment by KirillOsenkov** [comment]
"equivalent to the MSBuild.exe /bl command-line switch"

YAAS PLEASE

**Comment by tmeschter** [comment]
I like the idea of being able to turn this on and off via the UI, but it's not a project-level setting. Somewhere under Tools | Options would be more appropriate.

**Comment by KirillOsenkov** [comment]
Hmm, good point. I mean VS does build each project individually I think (??), so technically we could have it per project, but it's probably impractical to open each project and check/uncheck the box. Global is probably better.

Also what would be the default location of the binlog? Next to the .sln? Would the name be the default msbuild.binlog or should we allow to customize it?

**Comment by cartermp** [comment]
Yeah, makes sense. I guess also solution-level configuration could help, but I'm not sure which is the more discoverable place.

**Comment by tmeschter** [comment]
Within VS you might build the whole solution, or a specific project. My first thought is to name the files <solution_name>-<data_and_time>.binlog or <project_name>-<date_and_time>.binlog, respectively. I'm not sure about a default save location.

Also, this request has significant overlap with the Project System Tools extension since it can already capture these kinds of logs (and more).

**Comment by KirillOsenkov** [comment]
The moment you include date/time you'd need to think about a cleanup/expiration policy, otherwise these can flood the disk.
